# Python Guided Project #1 Submission

### Analzying Mobile App Data

In this guided project, Dataquest has us roleplaying as data analysts at an app building company. The apps that (our company) we build are free. The mission, if we choose to accept it:
        
>Our goal for this project is to 
>analyze data to help our developers 
>understand what type of apps are 
>likely to attract more users"
    


My goal in this project is to display my practiced skill and knowledge of the Jupyter Notebook environment and python.
    
**I'll do it. Twisted my arm.**

_as I go along I try to keep track of which questions I'm working on, above was Analzying Mobile App Data, 1/14, questions 1-2_

In [6]:
# function that they created for project
def explore_data(dataset, start, end, rows_and_cols=False):
    
    '''
    function that takes in dataset and explores it based on user's 
    set parameters.
    
    
    __parameters__
    
    dataset: (var to dataset) parameter to be explored
    start: (int) row number where you want the dataset to be explored from
    end: (int) row number where you want dataset to be explored to. 
    rows_and_cols: (bool) defaults to false, if made true, displays
    total rows and columns for entire dataset after exploration print
    out
    
    '''
    
    dataset_slice = dataset[start:end]
    
    for row in dataset_slice:
        print(row)
        print('\n') # adds newline when printing for prettiness
        
    if rows_and_cols:
        print('Number of rows:', len(dataset))
        print('Number of columns', len(dataset[0]))

### Opening and Exploring the Data

**(2/14) Questions 1-3**

In [2]:
# importing csv tools and opening datasets into namespace
# question 1

from csv import reader
applestoredata = list(reader(open('AppleStore.csv',encoding='utf8')))
googstoredata = list(reader(open('googleplaystore.csv',encoding='utf8')))

In [7]:
# question 2 
# exploring data with previously built function
explore_data(applestoredata,0,5,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns 16


q3)The structure of our Apple App Store data is 7198 rows and 16 columns. Columns that could assist us with our analysis are track_name (which is the app's name), price, reviews, installs, type, genre, and category. 

In [9]:
# q2
# number of rows and columns printed with the True parameter above this
# cell.
# exploring data with previously built function

explore_data(googstoredata,0,5,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns 13


q3) The structure of our Google Store App data is 10,842 rows and 13 columns. Less categories than the Apple store data. The column's that might be able to help us out with our analysis from this dataset would be like columns that we've previously found in the apple store dataset, app, category, installs, type, price, and genres.

In [10]:
explore_data(applestoredata,0,1)
explore_data(googstoredata,0,1)
# showing differences in categorical data

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']




**q3**
The documentation shows up as a 404, so I will not be able to post a reliable link for my project.

### Deleting Wrong Data

(3/14) 
    The Google Play data set has a dedicated discussion section, and we can see that one of the discussions describes an error for a certain row.

    1) Read the discussion and find out what the index of the row is.

    2) Print the row at that index to check whether it's indeed incorrect. Take into account the user reporting the error might or might have not removed the header row, so the index number might vary.

    3) If the row has an error, remove the row using the del statement. For instance, to remove the row with the index 149 from a data set data that is stored as a list of list, you can use the code del data[149].
    
    4) Make sure you don't run the del statement more than once, otherwise you'll delete more than one row.

    5) Read the discussion section for the App Store data set, and see whether you can find any reports of wrong data.

q1) went to [discussion section to find the bad row]('https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015')

In [21]:
# q2
print(googstoredata[0])
print(googstoredata[10473])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


The maximum a rating can be in the 'Rating' column can be 5 as said in the corresponding documentation and issue submitted to the dataset.

In [23]:
# q3 will be deleting this row of data with the del statement. highly dangerous. don't try at home without proper
# adult supervision
# q4, not running this cell more than twice

print(googstoredata[10473])
del googstoredata[10473]
print(googstoredata[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


**q5)** no wrong data rows in the apple store data csv according to some research through the [dataset's discussion]('https://www.kaggle.com/ramamet4/app-store-data-set-10k-apps/discussion') however did notice discussions in thread about duplicate rows which will be covered in the next section.

### Removing Duplicate Entries: Part One
(4/14)

    1) Using a combination of narrative and code, explain the reader that the Google Play data set has duplicate entries. Print a few duplicate rows to confirm.

    2) Count the number of duplicates using the technique we learned above.

    3) Explain that you won't remove the duplicates randomly. Describe the criterion you're going to use to remove the duplicates.

    4) We already suggested a criterion above, but you can come up with another criterion if you want. Make sure you support your criterion with at least one argument.

1) Through data collection and entry efforts we sometimes capture the same data multiple times or in ways that will obfuscate the dataset in unimaginable ways when it comes to building statistical understandings of the data.

In [28]:
for i in googstoredata:
    name = i[0]
    if name == 'Instagram':
        print(i)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [29]:
# 2) using the technique from the lessons
dup_apps = []
uniq_apps = []

for app in googstoredata:
    name = app[0]
    if name in uniq_apps:
        dup_apps.append(name)
    else:
        uniq_apps.append(name)
        
print("number of duplicate apps:", len(dup_apps))
print("number of unique apps:", len(uniq_apps))

number of duplicate apps: 1181
number of unique apps: 9660


q3 and 4) instead of removing the duplicates randomly, we can see that in the duplicates example that I printed above the number of reviews gets larger the closer to the bottom of the set. That means more recent data has been entered in the dataset and that we might be able to find this pattern elsewhere with duplicates.

to do this we'll build a dictionary that has a key as the app name and the value as the number of reviews. we'll use this to make sure that we're collecting the dataset entries with the highest number of reviews and then we'll use the dictionary as a set since we'll only have set of the app and it's reviews.

### Removing Duplicate Entries: Part Two
(5/14)
Create a dictionary where each key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app.

Start by creating an [click here for mission instructions]('https://app.dataquest.io/m/350/guided-project%3A-profitable-app-profiles-for-the-app-store-and-google-play-markets/5/removing-duplicate-entries-part-two') 

In [39]:
reviews_max = {}

for app in googstoredata:
    name = app[0]
    n_reviews = app[3]
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

what we are doing above is this:
    creating an empty dictionary
    initialize a for loop over the google store data
    initialize the name column to `name`
    if that name **is** in dictionary **but** the review number is lower than the current entry, replace with new entry
    else if the name is **not** found in the dictionary, add a new entry with the value being number of reviews.

to prove that we've reduced the number of duplicates down we can do some basic arithmatic.

In [44]:
len(googstoredata) - len(reviews_max)

1181

then we'll use the just made dictionary to spot and clean the duplicates out of another list we'll create.

In [48]:
goog_clean = []
goog_added = []

for app in googstoredata:
    name = app[0]
    n_reviews = app[3]
    
    if (reviews_max[name] == n_reviews) and (name not in goog_added):
        goog_clean.append(app)
        goog_added.append(name)

### Removing Non-English Apps: Part 1

    1) Write a function that takes in a string and returns False if there's any character in the string that doesn't belong to the set of common English characters, otherwise it returns True.

    -Inside the function, iterate over the input string. For each iteration check whether the number associated with the character is greater than 127. When a character is greater than 127, the function should immediately return False — the app name is probably non-English since it contains a character that doesn't belong to the set of common English characters.
    -If the loop finishes running without the return statement being executed, then it means no character had a corresponding number over 127 — the app name is probably English, so the functions should return True.
    2) Use your function to check whether these app names are detected as English or non-English:

    'Instagram'
    '爱奇艺PPS -《欢乐颂2》电视剧热播'
    'Docs To Go™ Free Office Suite'
    'Instachat 😜'
    
    

In [63]:
def is_english(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True

This function will help us map the characters in the dataset that are not english characters, or above a certain 'ascii' level, and any level above '127' is above the english character count.

We will actually use it twice in two different for clauses to clean out the lists of applications below.

### Removing Non-English Apps: Part Two

In [69]:
goog_engl = []
appl_engl = []

for app in goog_clean:
    name = app[0]
    if is_english(name):
        goog_engl.append(app)
        
for app in applestoredata:
    name = app[0]
    if is_english(name):
        appl_engl.append(app)

In [72]:
explore_data(goog_engl,2000,2005)

['Coupon Sherpa', 'SHOPPING', '4.4', '7793', '6.1M', '500,000+', 'Free', '0', 'Everyone', 'Shopping', 'February 6, 2018', '3.0.3', '4.0.3 and up']


['Gyft - Mobile Gift Card Wallet', 'SHOPPING', '4.1', '9701', '14M', '500,000+', 'Free', '0', 'Everyone', 'Shopping', 'July 25, 2018', '2.4.0', '4.4 and up']


['SavingStar - Grocery Coupons', 'SHOPPING', '4.2', '31519', '21M', '1,000,000+', 'Free', '0', 'Everyone', 'Shopping', 'June 12, 2018', '4.9.1', '4.0 and up']


['The Coupons App', 'SHOPPING', '4.5', '181990', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Shopping', 'April 27, 2018', 'Varies with device', 'Varies with device']


['Shopkick: Free Gift Cards, Shop Rewards & Deals', 'SHOPPING', '4.3', '213735', '43M', '10,000,000+', 'Free', '0', 'Everyone', 'Shopping', 'July 24, 2018', '5.3.73', '4.4 and up']




### Isolating the Free Apps

In [77]:
android_final = []
ios_final = []

for app in goog_engl:
    price = app[7]
    if price == '0':
        android_final.append(app)

for app in appl_engl:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)

In [80]:
print(len(android_final),len(ios_final))

8862 4056


In the above code cell I take the lists that have been cleaned of all non english terms and check to see what price they are. If they are free (cost of zero), the entries are then added to a last list.

### Most Common Apps by Genre: Part 1

    1) Give readers more context into why we want to find an app profile that fits both the App Store and Google Play. Explain our validation strategy for an app idea.

    2) Inspect both data sets and identify the columns you could use to generate frequency tables to find out what are the most common genres in each market.

Since we're a company that makes both android and apple applications we are looking for apps that have similar traffic over both platforms. Our validation strategy is that we would rapidly develop an android application, get user research data from it, and if we find that the app is appropriately profitable, we then move forward and develop and app for the apple platform and add it to the iOS App Store. 

To understand both datasets we need to find commonalities that the datasets share.

### Most Common Apps by Genre: Part 2

Building functions to understand data between the datasets

In [93]:
def freq_table(dataset, index):
    
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentages = {}
    
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages

The above function generates a frequency table.

In [94]:
def display_table(dataset, index):
    
    table = freq_table(dataset, index)
    table_display = []
    
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

This function takes the output from the last made function and flips it into descending order. 

In [126]:
display_table(ios_final, )

999273718 : 0.02465483234714004
999263877 : 0.02465483234714004
999239164 : 0.02465483234714004
999183538 : 0.02465483234714004
998936876 : 0.02465483234714004
998560520 : 0.02465483234714004
998528159 : 0.02465483234714004
998424233 : 0.02465483234714004
998360879 : 0.02465483234714004
998286883 : 0.02465483234714004
998276924 : 0.02465483234714004
998253660 : 0.02465483234714004
997811612 : 0.02465483234714004
997715441 : 0.02465483234714004
997676587 : 0.02465483234714004
997362197 : 0.02465483234714004
997252057 : 0.02465483234714004
996866372 : 0.02465483234714004
996855098 : 0.02465483234714004
996802374 : 0.02465483234714004
996509117 : 0.02465483234714004
996402903 : 0.02465483234714004
996345745 : 0.02465483234714004
996246479 : 0.02465483234714004
995999703 : 0.02465483234714004
995922287 : 0.02465483234714004
995125257 : 0.02465483234714004
995112030 : 0.02465483234714004
995014054 : 0.02465483234714004
994933845 : 0.02465483234714004
994905763 : 0.02465483234714004
99485580

953923489 : 0.02465483234714004
953917544 : 0.02465483234714004
953831664 : 0.02465483234714004
953511734 : 0.02465483234714004
953454701 : 0.02465483234714004
953036666 : 0.02465483234714004
952877179 : 0.02465483234714004
952871251 : 0.02465483234714004
952715194 : 0.02465483234714004
952694580 : 0.02465483234714004
952658839 : 0.02465483234714004
952578897 : 0.02465483234714004
952561811 : 0.02465483234714004
951937596 : 0.02465483234714004
951761438 : 0.02465483234714004
951744068 : 0.02465483234714004
951704333 : 0.02465483234714004
951627425 : 0.02465483234714004
951615217 : 0.02465483234714004
951536298 : 0.02465483234714004
951417499 : 0.02465483234714004
951364656 : 0.02465483234714004
951346423 : 0.02465483234714004
950984120 : 0.02465483234714004
950902520 : 0.02465483234714004
950795722 : 0.02465483234714004
950669893 : 0.02465483234714004
950593959 : 0.02465483234714004
950574221 : 0.02465483234714004
950527505 : 0.02465483234714004
950503358 : 0.02465483234714004
95002242

872040692 : 0.02465483234714004
871809581 : 0.02465483234714004
871169596 : 0.02465483234714004
871095743 : 0.02465483234714004
870099476 : 0.02465483234714004
869757898 : 0.02465483234714004
869231055 : 0.02465483234714004
868897264 : 0.02465483234714004
868759616 : 0.02465483234714004
868077558 : 0.02465483234714004
868013618 : 0.02465483234714004
867964741 : 0.02465483234714004
867887231 : 0.02465483234714004
867833019 : 0.02465483234714004
867827909 : 0.02465483234714004
867427452 : 0.02465483234714004
866617777 : 0.02465483234714004
865958296 : 0.02465483234714004
863998476 : 0.02465483234714004
863960527 : 0.02465483234714004
863882795 : 0.02465483234714004
863844475 : 0.02465483234714004
862863329 : 0.02465483234714004
862550306 : 0.02465483234714004
862303071 : 0.02465483234714004
861891048 : 0.02465483234714004
860822992 : 0.02465483234714004
860380109 : 0.02465483234714004
860011430 : 0.02465483234714004
859204347 : 0.02465483234714004
858919777 : 0.02465483234714004
85562788

668515245 : 0.02465483234714004
668159524 : 0.02465483234714004
667728512 : 0.02465483234714004
666575946 : 0.02465483234714004
665873523 : 0.02465483234714004
665795633 : 0.02465483234714004
664939913 : 0.02465483234714004
664575829 : 0.02465483234714004
664457128 : 0.02465483234714004
664425773 : 0.02465483234714004
663965632 : 0.02465483234714004
662749248 : 0.02465483234714004
662001778 : 0.02465483234714004
661149814 : 0.02465483234714004
660842333 : 0.02465483234714004
660653351 : 0.02465483234714004
659563061 : 0.02465483234714004
659425518 : 0.02465483234714004
659283309 : 0.02465483234714004
658785238 : 0.02465483234714004
658293394 : 0.02465483234714004
658257516 : 0.02465483234714004
657675944 : 0.02465483234714004
657500465 : 0.02465483234714004
657189652 : 0.02465483234714004
656971078 : 0.02465483234714004
656176278 : 0.02465483234714004
654810212 : 0.02465483234714004
653967729 : 0.02465483234714004
653508448 : 0.02465483234714004
653384976 : 0.02465483234714004
65329868

538248967 : 0.02465483234714004
538212549 : 0.02465483234714004
537452201 : 0.02465483234714004
537263603 : 0.02465483234714004
535886823 : 0.02465483234714004
535811906 : 0.02465483234714004
535640259 : 0.02465483234714004
535609266 : 0.02465483234714004
535500008 : 0.02465483234714004
535338086 : 0.02465483234714004
535153636 : 0.02465483234714004
534568162 : 0.02465483234714004
534539160 : 0.02465483234714004
534342529 : 0.02465483234714004
534174529 : 0.02465483234714004
533893269 : 0.02465483234714004
533451786 : 0.02465483234714004
533173905 : 0.02465483234714004
533079551 : 0.02465483234714004
533055738 : 0.02465483234714004
531761928 : 0.02465483234714004
531325150 : 0.02465483234714004
531184261 : 0.02465483234714004
530957474 : 0.02465483234714004
530621395 : 0.02465483234714004
530168168 : 0.02465483234714004
530003802 : 0.02465483234714004
530001625 : 0.02465483234714004
529997671 : 0.02465483234714004
529652920 : 0.02465483234714004
529479190 : 0.02465483234714004
52944360

310633997 : 0.02465483234714004
309735670 : 0.02465483234714004
309465525 : 0.02465483234714004
309187846 : 0.02465483234714004
309172177 : 0.02465483234714004
309025938 : 0.02465483234714004
308834491 : 0.02465483234714004
308750436 : 0.02465483234714004
307906541 : 0.02465483234714004
307764057 : 0.02465483234714004
307569751 : 0.02465483234714004
307386350 : 0.02465483234714004
307354030 : 0.02465483234714004
307184892 : 0.02465483234714004
306621789 : 0.02465483234714004
305939712 : 0.02465483234714004
305343404 : 0.02465483234714004
304878510 : 0.02465483234714004
303849934 : 0.02465483234714004
303191318 : 0.02465483234714004
302584613 : 0.02465483234714004
301724680 : 0.02465483234714004
301521403 : 0.02465483234714004
300704847 : 0.02465483234714004
300255638 : 0.02465483234714004
300238550 : 0.02465483234714004
300048137 : 0.02465483234714004
299853944 : 0.02465483234714004
299515267 : 0.02465483234714004
298867247 : 0.02465483234714004
298596830 : 0.02465483234714004
29760695

1135328912 : 0.02465483234714004
1135213693 : 0.02465483234714004
1135126803 : 0.02465483234714004
1135031533 : 0.02465483234714004
1134895689 : 0.02465483234714004
1134858124 : 0.02465483234714004
1134680712 : 0.02465483234714004
1134529728 : 0.02465483234714004
1134516647 : 0.02465483234714004
1134511982 : 0.02465483234714004
1134424705 : 0.02465483234714004
1134424698 : 0.02465483234714004
1134144728 : 0.02465483234714004
1134086133 : 0.02465483234714004
1133990005 : 0.02465483234714004
1133919651 : 0.02465483234714004
1133898987 : 0.02465483234714004
1133896041 : 0.02465483234714004
1133749572 : 0.02465483234714004
1133706938 : 0.02465483234714004
1133678984 : 0.02465483234714004
1133510322 : 0.02465483234714004
1133481001 : 0.02465483234714004
1133417566 : 0.02465483234714004
1133173046 : 0.02465483234714004
1133084307 : 0.02465483234714004
1133064144 : 0.02465483234714004
1133031538 : 0.02465483234714004
1132817034 : 0.02465483234714004
1132762804 : 0.02465483234714004
1132754743

1113413855 : 0.02465483234714004
1113359219 : 0.02465483234714004
1113304677 : 0.02465483234714004
1113223808 : 0.02465483234714004
1112989865 : 0.02465483234714004
1112922482 : 0.02465483234714004
1112858566 : 0.02465483234714004
1112826311 : 0.02465483234714004
1112500249 : 0.02465483234714004
1112481571 : 0.02465483234714004
1112477374 : 0.02465483234714004
1112382631 : 0.02465483234714004
1112350146 : 0.02465483234714004
1112307917 : 0.02465483234714004
1112162323 : 0.02465483234714004
1112081391 : 0.02465483234714004
1112071930 : 0.02465483234714004
1112071865 : 0.02465483234714004
1112052555 : 0.02465483234714004
1112040900 : 0.02465483234714004
1111876388 : 0.02465483234714004
1111665247 : 0.02465483234714004
1111487818 : 0.02465483234714004
1111428249 : 0.02465483234714004
1110700363 : 0.02465483234714004
1110657008 : 0.02465483234714004
1110480334 : 0.02465483234714004
1110368735 : 0.02465483234714004
1110286206 : 0.02465483234714004
1110273623 : 0.02465483234714004
1110261314

1092258126 : 0.02465483234714004
1092148948 : 0.02465483234714004
1092109109 : 0.02465483234714004
1091997580 : 0.02465483234714004
1091993742 : 0.02465483234714004
1091974564 : 0.02465483234714004
1091961831 : 0.02465483234714004
1091946582 : 0.02465483234714004
1091944550 : 0.02465483234714004
1091891746 : 0.02465483234714004
1091824117 : 0.02465483234714004
1091700242 : 0.02465483234714004
1091699305 : 0.02465483234714004
1091684426 : 0.02465483234714004
1091627644 : 0.02465483234714004
1091569546 : 0.02465483234714004
1091522492 : 0.02465483234714004
1091487691 : 0.02465483234714004
1091456207 : 0.02465483234714004
1091366179 : 0.02465483234714004
1091215595 : 0.02465483234714004
1091192778 : 0.02465483234714004
1091186897 : 0.02465483234714004
1091144704 : 0.02465483234714004
1091126272 : 0.02465483234714004
1090877424 : 0.02465483234714004
1090842524 : 0.02465483234714004
1090826925 : 0.02465483234714004
1090759947 : 0.02465483234714004
1090731387 : 0.02465483234714004
1090636013

1072007547 : 0.02465483234714004
1071976327 : 0.02465483234714004
1071974253 : 0.02465483234714004
1071913738 : 0.02465483234714004
1071770573 : 0.02465483234714004
1071770565 : 0.02465483234714004
1071767366 : 0.02465483234714004
1071521565 : 0.02465483234714004
1071511733 : 0.02465483234714004
1071497654 : 0.02465483234714004
1071426243 : 0.02465483234714004
1071366753 : 0.02465483234714004
1071310449 : 0.02465483234714004
1071224055 : 0.02465483234714004
1071077611 : 0.02465483234714004
1071045817 : 0.02465483234714004
1071039429 : 0.02465483234714004
1071021891 : 0.02465483234714004
1071006943 : 0.02465483234714004
1070999857 : 0.02465483234714004
1070864992 : 0.02465483234714004
1070855108 : 0.02465483234714004
1070725569 : 0.02465483234714004
1070709625 : 0.02465483234714004
1070702119 : 0.02465483234714004
1070670743 : 0.02465483234714004
1070615218 : 0.02465483234714004
1070525362 : 0.02465483234714004
1070506962 : 0.02465483234714004
1070492711 : 0.02465483234714004
1070255846

1055697824 : 0.02465483234714004
1055677337 : 0.02465483234714004
1055672607 : 0.02465483234714004
1055596969 : 0.02465483234714004
1055571047 : 0.02465483234714004
1055570687 : 0.02465483234714004
1055502792 : 0.02465483234714004
1055455919 : 0.02465483234714004
1055397979 : 0.02465483234714004
1055302647 : 0.02465483234714004
1055281310 : 0.02465483234714004
1055274701 : 0.02465483234714004
1054948424 : 0.02465483234714004
1054933075 : 0.02465483234714004
1054867839 : 0.02465483234714004
1054651434 : 0.02465483234714004
1054355818 : 0.02465483234714004
1054297141 : 0.02465483234714004
1054256307 : 0.02465483234714004
1054033219 : 0.02465483234714004
1054011814 : 0.02465483234714004
1053874290 : 0.02465483234714004
1053808624 : 0.02465483234714004
1053807892 : 0.02465483234714004
1053800699 : 0.02465483234714004
1053688442 : 0.02465483234714004
1053533457 : 0.02465483234714004
1053524307 : 0.02465483234714004
1053388964 : 0.02465483234714004
1053285387 : 0.02465483234714004
1053066296

1032240256 : 0.02465483234714004
1032040117 : 0.02465483234714004
1031867079 : 0.02465483234714004
1031857811 : 0.02465483234714004
1031838318 : 0.02465483234714004
1031728646 : 0.02465483234714004
1031706898 : 0.02465483234714004
1031704553 : 0.02465483234714004
1031569021 : 0.02465483234714004
1031391869 : 0.02465483234714004
1031228862 : 0.02465483234714004
1031164407 : 0.02465483234714004
1031152574 : 0.02465483234714004
1031002863 : 0.02465483234714004
1030815031 : 0.02465483234714004
1030804846 : 0.02465483234714004
1030634472 : 0.02465483234714004
1030505688 : 0.02465483234714004
1030487391 : 0.02465483234714004
1030447361 : 0.02465483234714004
1029375276 : 0.02465483234714004
1029273159 : 0.02465483234714004
1029033692 : 0.02465483234714004
1028984371 : 0.02465483234714004
1028939296 : 0.02465483234714004
1028902794 : 0.02465483234714004
1028464453 : 0.02465483234714004
1028362533 : 0.02465483234714004
1028003495 : 0.02465483234714004
1027688889 : 0.02465483234714004
1027352017

### Most Common Apps by Genre: Part 3

The heavy majority of app users are gamers. From there we're nearly descending equal parts everything else other than entertainment. What this means is that we have a list of the most downloaded free applications that use English in their interface and/or app details on the platform. 

### Most Popular Apps by Genre on the App Store



In [119]:
genres_ios = freq_table(ios_final, -5)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[-5]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 53078.195804195806
Photo & Video : 27249.892215568863
Games : 18924.68896765618
Music : 56482.02985074627
Reference : 67447.9
Health & Fitness : 19952.315789473683
Weather : 47220.93548387097
Utilities : 14010.100917431193
Travel : 20216.01785714286
Shopping : 18746.677685950413
News : 15892.724137931034
Navigation : 25972.05
Lifestyle : 8978.308510638299
Entertainment : 10822.961077844311
Food & Drink : 20179.093023255813
Sports : 20128.974683544304
Book : 8498.333333333334
Finance : 13522.261904761905
Education : 6266.333333333333
Productivity : 19053.887096774193
Business : 6367.8
Catalogs : 1779.5555555555557
Medical : 459.75


From the

### Most Popular Apps by Genre on Google Play

In [120]:
categories_android = freq_table(android_final,1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1820673.076923077
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15560965.599534342
FAMILY : 3694276.334922527
MEDICAL : 120616.48717948717
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17805627.643678162
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10682301.033377837
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

In [121]:
under_100_m = []

for app in android_final:
    n_installs = app[5]
    n_installs = n_installs.replace(',','')
    n_installs = n_installs.replace('+','')
    if (app[1] == 'COMMUNICATION') and (float(n_installs) < 100000000):
        under_100_m.append(float(n_installs))
        
sum(under_100_m) / len(under_100_m)

3603485.3884615386

In [122]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [124]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


In [125]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

In conclusion, the direction this analysis began to illuminate is that it would advantageous to experiment putting out popular books as a stand alone application.